### ISSUE TITLE
Watching multiple instances of same model class

### ISSUE THREAD

**Issue by tshrjn** [wandb/client#626]
`wandb --version && python --version && uname`

* Weights and Biases version: 0.8.13
* Python version: 3.7.4
* Operating System: Linux

### Description

Trying to watch multiple model instances of the same model class. For example:
```
model1 = ResNet18()
model2 = ResNet18()

wandb.watch(model1, log='all')
wandb.watch(model2, log='all')
```
And getting the following error on the line watching `model2`:
```
ValueError: A hook has already been set under name "gradients/layer1.0.weight"
```

This looks like is happening because `layer1.0` already exists. The gradients should also have the variable name of the model to resolve this issue.

Edit:
Complete Traceback:

```
Traceback (most recent call last):
  File "/home/tushar/.vscode-server/extensions/ms-python.python-2019.10.41019/pythonFiles/ptvsd_launcher.py", line 43, in <module>
    main(ptvsdArgs)
  File "/home/tushar/.vscode-server/extensions/ms-python.python-2019.10.41019/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 432, in main
    run()
  File "/home/tushar/.vscode-server/extensions/ms-python.python-2019.10.41019/pythonFiles/lib/python/old_ptvsd/ptvsd/__main__.py", line 316, in run_file
    runpy.run_path(target, run_name='__main__')
  File "/home/tushar/anaconda3/envs/pytorch_env/lib/python3.7/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/home/tushar/anaconda3/envs/pytorch_env/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/home/tushar/anaconda3/envs/pytorch_env/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/tushar/some_py_torch_project/train.py", line 67, in <module>
    wandb.watch(encoder2, log='all')
  File "/home/tushar/anaconda3/envs/3d_mesh/lib/python3.7/site-packages/wandb/__init__.py", line 143, in watch
    model, log_parameters=log_parameters, log_gradients=log_gradients, prefix=prefix, log_freq=log_freq)
  File "/home/tushar/anaconda3/envs/pytorch_env/lib/python3.7/site-packages/wandb/wandb_torch.py", line 112, in add_log_hooks_to_pytorch_module
    parameter, 'gradients/' + prefix + name, log_track_grad)
  File "/home/tushar/anaconda3/envs/pytorch_env/lib/python3.7/site-packages/wandb/wandb_torch.py", line 196, in _hook_variable_gradient_stats
    'A hook has already been set under name "{}"'.format(name))
ValueError: A hook has already been set under name "gradients/layer1.0.weight"
```

**Comment by raubitsj** [comment]
@tshrjn: thank you for this bug report we just added the ability to call wandb.watch multiple times in a run and it looks like there is an issue with namespaces.  If you use the old behavior of running:
```python
wandb.watch([model1, model2], log='all')
```

Then your program will not crash

**Comment by tshrjn** [comment]
Thanks a ton for your quick response.

That did the trick! 👍 

Also, In my opinion having a name argument would for [`watch`](https://github.com/wandb/client/blob/master/wandb/__init__.py#L107) function would be a good idea since anyway, `add_log_hooks_to_pytorch_module` in [wandb_torch.py], which [called](https://github.com/wandb/client/blob/master/wandb/__init__.py#L142) for 
 this purpose, (https://github.com/wandb/client/blob/master/wandb/wandb_torch.py#L82) already has the name argument to be used as a prefix.

**Comment by vanpelt** [comment]
Hey @tshrjn we just released 0.8.14 which fixes the case of calling wandb.watch() multiple times.  We didn't add a name propery yet, but likely will in a future release.
